### PR TITLE
Make injection namespace check multicluster aware

### PR DESCRIPTION
**Pull request by howardjohn**
I think the current logic is not quite right for a case where we inject cross-cluster.

This builds on https://github.com/istio/istio/pull/49260, just look at the last commit for now.

### CODE REVIEW

**Comment by howardjohn** [comment]
> I think this may break existing behavior, previously we depend on namespaces in config cluster.
> 
> maybe we should add a feature flag

This is only impacting OpenShift users and the openshift maintainers (@jwendell) recommended this. If a user had mismatched namespaces in this way, they would be broken. I do not think it is possible for this PR to break a user, only fix them. If I am misunderstanding some case, happy to add a flag.

**Comment by hzxuzhonghu** [comment]
@howardjohn This not only influence redhat, the external controlplane mode also suffer from this change.  This multi clusters 
 mode only have one control plane deployed. 
https://istio.io/latest/docs/setup/install/external-controlplane/#adding-clusters

This change may break this kind of users, better to have a flag.

**Comment by howardjohn** [comment]
It only impacts OpenShift, see https://github.com/istio/istio/pull/49336/files#diff-e57c2ab4e6af0f24e7285fad8ec9fe8b688fad1350b6d6c405f75a067d65e0dcR210. Without this being `true` none of this code will ever execute

**Comment by hzxuzhonghu** [comment]
Oh, I see the namespace selector is set in mutatingwebhookConfiguration.

**Comment by hzxuzhonghu** [comment]
LGTM defer to @jwendell

File: pkg/kube/inject/webhook.go
Comment (jwendell) [inline pkg/kube/inject/webhook.go:209]: Do we still need this check?

**Review by jwendell — COMMENTED** [review]
LGTM, just a minor comment

File: pkg/kube/inject/webhook.go
Comment (howardjohn) [reply pkg/kube/inject/webhook.go:209]: I don't think this PR changes this. I think it can still be nil in (very niche) cases where we don't have a k8s connection at all

Comment (hzxuzhonghu) [reply pkg/kube/inject/webhook.go:209]: This can be removed in another pr

**Review by hzxuzhonghu — APPROVED** [review]

**Comment by dgn** [comment]
looks like this PR broke OpenShift support- I'm not fully following, but it seems to add a `MultiCluster` field that is never initialized and only accessed if `IsOpenShift()` returns true?

**Comment by howardjohn** [comment]
Thanks @dgn !

**Pull request merged**
